### PR TITLE
Managed identity in InvokeUploadModuleLogs function

### PR DESCRIPTION
**Pull request by eedorenko**
- If HubConnectionString is not specified, then use managed identity to connect to Hub
- When using managed identity, then HubHostName env variable should be specified (e.g. iotstarter-iothub.azure-devices.net)

### CODE REVIEW

File: FunctionApp/FunctionApp/InvokeUploadModuleLogs.cs [hunk 1]
```diff
@@ -60,6 +63,10 @@ public static class InvokeUploadModuleLogs
                     _logsContentType = "json";
                 #endregion
 
+                TokenCredential tokenCredential = string.IsNullOrEmpty(_iotHubConnectionString) ? 
+                    tokenCredential = new DefaultAzureCredential() : null;
+
```
Comment (Ioana37) [inline FunctionApp/FunctionApp/InvokeUploadModuleLogs.cs]: Nit: Please remove empty lines at 68 and 101.

File: FunctionApp/FunctionApp/InvokeUploadModuleLogs.cs [hunk 2]
```diff
@@ -87,11 +94,16 @@ public static class InvokeUploadModuleLogs
                 }
                 else
                 {
-                    // query IoT edge devices
-                    var registryManager = RegistryManager.CreateFromConnectionString(_iotHubConnectionString);
-                    var query = registryManager.CreateQuery(_iotDeviceQuery);
-                    var devices = (await query.GetNextAsJsonAsync()).ToArray();
-                    deviceIds = devices.Select(x => JsonConvert.DeserializeObject<JObject>(x).GetValue("deviceId").ToString()).ToArray();
+                    // query IoT edge devices                    
+                    using  (var registryManager = string.IsNullOrEmpty(_iotHubConnectionString) ?
```
Comment (Ioana37) [inline FunctionApp/FunctionApp/InvokeUploadModuleLogs.cs]: Looks good, but out of curiosity, why did you add a 'using'?

**Review by Ioana37 — APPROVED** [review]
Looks nice, just minor comments :)

File: FunctionApp/FunctionApp/InvokeUploadModuleLogs.cs (hunk 2 above)
Comment (eedorenko) [reply FunctionApp/FunctionApp/InvokeUploadModuleLogs.cs]: Because it's a disposable object. [To get Dispose invoked correctly](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement#remarks) once I don't need registry manager anymore. [Azure sample](https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/4d52e622d6b36eff1fa70064dcd40874c94fbc53/iot-hub/Samples/service/RoleBasedAuthenticationSample/Program.cs#L48) also uses "using" with ServiceClient. Java has a similar concept [AutoClosable](https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html).

**Review by Ioana37 — APPROVED** [review]
